components/dashboard/src/components/RepositoryFinder.tsx

```diff
@@ -4,137 +4,96 @@
-    const onKeyDown = (event: React.KeyboardEvent) => {
-        if (!selectedSearchResult) {
-            return;
-        }
-        const selectedIndex = searchResults.indexOf(selectedSearchResult);
-        const select = (index: number) => {
-            // Implement a true modulus in order to "wrap around" (e.g. `select(-1)` should select the last result)
-            // Source: https://stackoverflow.com/a/4467559/3461173
-            const n = Math.min(searchResults.length, MAX_DISPLAYED_ITEMS);
-            setSelectedSearchResult(searchResults[((index % n) + n) % n]);
-        };
-        if (event.key === "ArrowDown") {
-            event.preventDefault();
-            select(selectedIndex + 1);
-            return;
-        }
-        if (event.key === "ArrowUp") {
-            event.preventDefault();
-            select(selectedIndex - 1);
-            return;
-        }
-    };
+interface RepositoryFinderProps {
+    initialValue?: string;
+    maxDisplayItems?: number;
+    setSelection: (selection: string) => void;
+}
 
+export default function RepositoryFinder(props: RepositoryFinderProps) {
+    const [suggestedContextURLs, setSuggestedContextURLs] = useState<string[]>(loadSearchData());
     useEffect(() => {
-        const element = document.querySelector(`a[href='/#${selectedSearchResult}']`);
-        if (element) {
-            element.scrollIntoView({ behavior: "smooth", block: "nearest" });
-        }
-    }, [selectedSearchResult]);
-
-    const onSubmit = (event: React.FormEvent) => {
-        event.preventDefault();
-        if (selectedSearchResult) {
-            window.location.href = "/#" + selectedSearchResult;
-        }
-    };
+        getGitpodService()
+            .server.getSuggestedContextURLs()
+            .then((urls) => {
+                setSuggestedContextURLs(urls);
+                saveSearchData(urls);
+            });
+    }, [suggestedContextURLs]);
+
+    const getElements = useMemo(
+        () => (searchString: string) => {
+            const result = [...suggestedContextURLs];
+            try {
+                // If the searchString is a URL, and it's not present in the proposed results, "artificially" add it here.
+                new URL(searchString);
+                if (!result.includes(searchString)) {
+                    result.push(searchString);
+                }
+            } catch {}
+            return result
+                .filter((e) => e.toLowerCase().indexOf(searchString.toLowerCase()) !== -1)
+                .map(
+                    (e) =>
+                        ({
+                            id: e,
+                            element: (
+                                <div className="flex-col ml-1 mt-1 flex-grow">
+                                    <div className="flex">
+                                        <div className="text-gray-700 dark:text-gray-300 font-semibold">
+                                            {e.substring(e.indexOf("//") + 2)}
```

Comment:
A comment here would be helpful to understand what this is for.  I think it strips the protocol off the value?